src/storage/txn/process.rs

```diff
@@ -451,7 +451,8 @@ fn process_read_impl<E: Engine>(
                 Ok(ProcessResult::Res)
             } else {
                 let next_scan_key = if has_remain {
-                    // There might be more locks.
+                    // Locks is resolved via drop (`_lock`) here.
```

Comment:
Does this means the `ResolveLock` operations resolves locks that were scanned from the db when dropping `_lock`?
This seems to be not true. `ResolveLock` is processed cycle by cycle, and in each cycle it goes to `process_read` (here), scan a set of locks, and then goes to `process_write` to delete these scanned locks. `scan_key` marks where the next cycle should start from, and `_lock` is not useful here so it is dropped.

Reply:
Oh, okay. Thank you for clarifying.